### PR TITLE
Log bad request error details during dev

### PR DESCRIPTION
**Pull request by mrbbot**
Currently, when the edge preview service returns a 400 Bad Request error during a `wrangler dev` session because the user's worker is invalid, all you see is `Error: HTTP status client error (400 Bad Request) for url`. When calling `wrangler publish`, you are given the actual reason, for example:
```shell
Error: Something went wrong with the request to Cloudflare...
No event handlers were registered. This script does nothing.
 [API code: 10021]
```
These errors also terminate the process, meaning you have to restart `wrangler dev` each time.

This PR replaces the error text for 400 responses with the actual reason. If an error occurs when watching files, it is logged, but the watcher remains active, so the user is able to fix the issue and wrangler can try upload again:
```js
addEventListener("bad", (event) => {
  event.respondWith(new Response());
});
```
```shell
$ wrangler dev      
Error: Something went wrong with the request to Cloudflare...
No event handlers were registered. This script does nothing.
 [API code: 10021]

# changing "bad" to "fetch"
$ wrangler dev
💁  watching "./"
👂  Listening on http://127.0.0.1:8787
🌀  Detected changes... # changing "fetch" to "bad"
⚠️ Something went wrong with the request to Cloudflare...
No event handlers were registered. This script does nothing.
 [API code: 10021]
🌀  Detected changes... # changing "bad" to "fetch"
```

### CODE REVIEW

File: src/commands/dev/edge/setup.rs
Comment (jyn514) [inline src/commands/dev/edge/setup.rs:56]:
```suggestion
    if response.status() == StatusCode::BAD_REQUEST {
```

File: src/commands/dev/edge/watch.rs
Comment (jyn514) [inline src/commands/dev/edge/watch.rs:48]: Someone else should review this bit, I'm not sure if the cloudflare API could ever send back a 400 if the session needs to be refreshed. The rest of the PR looks good though :)

Comment (caass) [reply src/commands/dev/edge/watch.rs:48]: In my quick testing, I didn't notice this causing any issues -- I'm gonna merge this, and if it winds up being problematic we can follow up with a bugfix PR

**Review by caass — APPROVED** [review]

**Pull request merged**
